podvm/qcow2/misc-settings.sh

```diff
@@ -36,8 +36,86 @@ then
     *)
         ;;
     esac
-# else if...
-#
+fi
+
+# Setup oneshot systemd service for AWS and Azure to enable NAT rules
+if [ "$CLOUD_PROVIDER" == "azure" ] || [ "$CLOUD_PROVIDER" == "aws" ]
+then
+# Create a oneshot systemd service to execute setup-nat-for-imds.sh
+# during first boot
+cat <<END > /etc/systemd/system/setup-nat-for-imds.service
+[Unit]
+Description=Setup NAT for IMDS
+After=kata-agent.service
+Wants=kata-agent.service
+Type=oneshot
+
+[Service]
+ExecStart=/usr/local/bin/setup-nat-for-imds.sh
+
+[Install]
+WantedBy=multi-user.target
+END
+
+# Add a link to the service in multi-user.target.wants
+ln -s /etc/systemd/system/setup-nat-for-imds.service /etc/systemd/system/multi-user.target.wants/setup-nat-for-imds.service
+
+# Create a script to setup NAT for IMDS
+cat <<END > /usr/local/bin/setup-nat-for-imds.sh
+#!/bin/bash
+# This script sets up NAT for IMDS on Azure/AWS
+# This is required for IMDS to work on Azure/AWS
+# This script is executed as a oneshot systemd service
+# during first boot
+# Define variables
+POD_NS="podns"
+IMDS_IP_CIDR="169.254.169.254/32"
+VETH_NET_CIDR="192.168.100.0/30"
+VETH1_IP_CIDR="192.168.100.1/30"
+VETH2_IP_CIDR="192.168.100.2/30"
+VETH_GW="192.168.100.1"
+
+
+# trap errors
+trap 'echo "Error: \$0:\$LINENO stopped"; exit 1' ERR INT
+
+# Function to setup veth pair
+function setup_veth() {
+    # Set up a veth pair
+    ip link add veth2 type veth peer name veth1
+    # Move the veth endpoint to pod namespace
+    ip link set veth2 netns \${POD_NS}
+        # Assign IP on the host endpoint of the pair an IP and bring it up
```

Comment:
> personally for discoverability, i'd prefer the script to be in `files/usr/local/bin/setup-nat-for-imds.service` and possible the oneshot unit file in `files/etc/systemd/system` too. A custom `misc-settings.sh` in /azure and /aws would then just invoke `systemctl enable setup-nas-for-imds`.

Yeah that's a possibility as well. Just that I wanted to keep the things consistent between podvm/aws/azure/ibmcloud etc w.r.to image customisation files, so that removing all these duplicates as part of https://github.com/confidential-containers/cloud-api-adaptor/issues/899 is relatively easier.